src/chessboard/hooks/useArrows.ts

```diff
@@ -79,6 +88,7 @@ export const useArrows = (
   return {
     arrows: toArray(arrows),
     newArrow,
+    isCurrentlyDrawingArrow,
```

Comment:
Technically you can avoid creating a new `isCurrentlyDrawingArrow` state variable, and use `newArrow` for getting the value you need.
 
```suggestion
    isCurrentlyDrawingArrow: Boolean(newArrow),
```

Reply:
As I explained in the initial PR, it is possible for newArrow to be undefined, but still have isCurrentlyDrawingArrow to be true. This covers the instance when the user begins drawing an arrow, then returns the mouse to the original square without letting go of right-click.

Reply:
By adding the suggestion and casting isCurrentlyDrawingArrow to bool check, the onRightClickUp is now always fired when ending an arrow drag on the same square. This is the original behavior. 

https://github.com/Clariity/react-chessboard/assets/20730722/81908613-af9c-4127-bf6f-59c0eae77c4b